File: backend/src/main/java/com/example/backend/domain/category/entity/Category.java
package com.example.backend.domain.category.entity;


import com.example.backend.domain.groupcategory.GroupCategory;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "\"categories\"")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String name;

    @Column
    @Enumerated(value = EnumType.STRING)
    private CategoryType categoryType;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<GroupCategory> groupCategories = new ArrayList<>();

    @Builder
    public Category(String name, CategoryType categoryType) {
        this.name = name;
        this.categoryType = categoryType;
    }

    public void modify(String name, CategoryType categoryType) {
        this.name = name;
        this.categoryType = categoryType;
    }
}

File: backend/src/main/java/com/example/backend/domain/category/entity/Category.kt
package com.example.backend.domain.category.entity

import com.example.backend.domain.groupcategory.GroupCategory
import jakarta.persistence.*
import java.util.ArrayList

@Entity
@Table(name = "categories")
class Category(
    @Column var name: String,

    @Enumerated(EnumType.STRING)
    @Column var categoryType: CategoryType
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @OneToMany(mappedBy = "category", cascade = [CascadeType.ALL], orphanRemoval = true)
    val groupCategories: MutableList<GroupCategory> = ArrayList()

    fun modify(name: String, categoryType: CategoryType) {
        this.name = name
        this.categoryType = categoryType
    }
}
